chain/transaction_test.go

```diff
@@ -130,6 +200,13 @@ func TestMarshalUnmarshal(t *testing.T) {
 	for i, action := range signedTx.Actions {
 		require.Equal(tx.Actions[i], action)
 	}
+	// validating tx ID, Bytes and Size
+	writerPacker := codec.NewWriter(0, consts.NetworkSizeLimit)
+	err = signedTx.Marshal(writerPacker)
+	require.NoError(err)
+	require.Equal(signedTx.ID(), utils.ToID(writerPacker.Bytes()))
+	require.Equal(signedTx.Bytes(), writerPacker.Bytes())
+	require.Len(writerPacker.Bytes(), signedTx.Size())
```

Comment:
```suggestion
```

Can we remove the explicit length check right after we've already checked the bytes themselves are equivalent?

Reply:
re: https://github.com/ava-labs/hypersdk/pull/1665/files#r1819411335 yes, we should add a check for unmarshalling the signed/unsigned tx bytes results in the identical `TxData` / `Transaction`